src/cio_websocket.c

```diff
@@ -70,6 +70,37 @@ static void remove_websocket_header(const struct cio_websocket_write_job *job)
 	cio_write_buffer_queue_dequeue(job->wbh);
 }
 
+static inline void swap(uint8_t mask[4], uint_fast8_t first, uint_fast8_t last)
+{
+	uint8_t tmp = mask[first];
+	mask[first] = mask[last];
+	mask[last] = tmp;
+}
+
+static inline void rotate(uint8_t mask[4], uint_fast8_t middle)
+{
+	uint_fast8_t first = 0;
+	uint_fast8_t last = 4;
+	uint_fast8_t next = middle;
+
+	while (first != next) {
+		swap(mask, first++, next++);
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Extract the assignment from this expression. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=c%3AAssignmentInSubExpression)
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Extract the assignment from this expression. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=c%3AAssignmentInSubExpression)